cmd/checkpoint/main.go

```diff
@@ -505,14 +545,85 @@ func (c *kubeletClient) localParentPods() map[string]*v1.Pod {
 	return podListToParentPods(podList)
 }
 
-// localRunningPods uses the /runningpods/ kubelet api to retrieve the local container runtime pod state
+// localRunningPods uses the CRI shim to retrieve the local container runtime pod state
 func (c *kubeletClient) localRunningPods() map[string]*v1.Pod {
-	podList := new(v1.PodList)
-	if err := c.secureClient.Get().AbsPath("/runningpods/").Do().Into(podList); err != nil {
-		// Assume there are no local running pods.
-		glog.Errorf("failed to list local running pods, assuming none are ready: %v", err)
+
+	pods := make(map[string]*v1.Pod)
+
+	sandboxes, err := c.getKubeletSandboxes(false)
+	if err != nil {
+		glog.Errorf("failed to list running sandboxes: %v", err)
+		return nil
+	}
+
+	// Add all pods that sandboxes are apart of
+	for _, s := range sandboxes {
+		if s.Metadata == nil {
+			glog.V(4).Infof("Sanbox does not have metadata: %+v", s)
+			continue
+		}
+
+		podName := s.Metadata.Namespace + "/" + s.Metadata.Name
+		if _, ok := pods[podName]; !ok {
+			p := &v1.Pod{}
+			p.UID = types.UID(s.Metadata.Uid)
+			p.Name = s.Metadata.Name
+			p.Namespace = s.Metadata.Namespace
+
+			pods[podName] = p
+		}
+	}
+
+	containers, err := c.getKubeletContainers(false)
+	if err != nil {
+		glog.Errorf("failed to list running containers: %v", err)
+		return nil
+	}
+
+	// Add all pods that containers are apart of
+	for _, c := range containers {
+		if c.Metadata == nil {
+			glog.V(4).Infof("Container does not have metadata: %+v", c)
+			continue
+		}
+
+		podName := c.Labels[kubelettypes.KubernetesPodNamespaceLabel] + "/" + c.Labels[kubelettypes.KubernetesPodNameLabel]
+		if _, ok := pods[podName]; !ok {
+			p := &v1.Pod{}
+			p.UID = types.UID(c.Labels[kubelettypes.KubernetesPodUIDLabel])
+			p.Name = c.Labels[kubelettypes.KubernetesPodNameLabel]
+			p.Namespace = c.Labels[kubelettypes.KubernetesPodNamespaceLabel]
+
+			pods[podName] = p
+		}
+	}
+
+	return pods
+}
+
+func (c *kubeletClient) getKubeletContainers(all bool) ([]*runtimeapi.Container, error) {
```

Comment:
nit: `all` isn't used

Reply:
It's used 4 lines down

`if !all {`

Reply:
I think aaron means we always call with `false` so the boolean is redundant?
I am fine with leave it as is tho.